lib/src/front_end/ast_node_visitor.dart

```diff
@@ -667,7 +667,13 @@ class AstNodeVisitor extends ThrowingAstVisitor<void>
 
   @override
   void visitListLiteral(ListLiteral node) {
-    createCollection(node, node.leftBracket, node.elements, node.rightBracket);
+    createCollection(
+      node.constKeyword,
+      node.leftBracket,
+      node.elements,
+      node.rightBracket,
+      typeArguments: node.typeArguments,
```

Comment:
Nit: I'd move the type arguments argument up to reflect the order that these appear in the source syntax:

```dart
    createCollection(
      node.constKeyword,
      typeArguments: node.typeArguments,
      node.leftBracket,
      node.elements,
      node.rightBracket,
  );
```